infrastructure-playbooks/recover-osds-after-ssd-journal-failure.yml

```diff
@@ -0,0 +1,124 @@
+# @param index:  The index of this partition
+# @param size:  The size of this partition
+# @param osd_id: Which osds's journal this parttition for.
+#
+# ansible-playbook recover-osds-after-ssd-journal-failure.yml
+#     Prompts for select which host to recover, defaults to null,  
+#     doesn't select host the recover ssd. Input the hostname
+#     which to recover osds after ssd journal failure
+#
+# ansible-playbook -e target_host=hostname \
+#     recover-osds-after-ssd-journal-failure.yml
+#     Overrides the prompt using -e option. Can be used in
+#     automation scripts to avoid interactive prompt.
+
+- hosts: localhost
+  gather_facts: no
+  vars_prompt:
+  - name: target_host
+    prompt: please enter the target hostname which to recover osds after ssd journal failure
+    private: no
+  tasks:
+    - add_host:
+        name: "{{ target_host }}"
+        groups: dynamically_created_hosts
+
+- hosts: dynamically_created_hosts
+  vars:
+   journal_typecode: 45b0969e-9b03-4f30-b4c6-b4b80ceff106
+   dev_ssds: []
+
+  tasks:
+  - fail: msg="please define dev_ssds variable"
+    when: dev_ssds|length <= 0
+
+  - name: get the name of the existing ceph cluster
+    shell: |
+      basename $(grep -R fsid /etc/ceph/ | egrep -o '^[^.]*')
+    changed_when: false
+    register: cluster_name
+        
+  - name: get osd(s) if directory stat
+    stat:
+      path: "/var/lib/ceph/osd/{{ cluster_name.stdout }}-{{ item.1.osd_id }}/journal_uuid"
+    register: osds_dir_stat
+    with_subelements:
+      - "{{ dev_ssds }}"
+      - partitions
+ 
+  - name: exit playbook osd(s) is not on this host
+    fail:
+        msg: exit playbook osds is not no this host
+    with_items:
+        osds_dir_stat.results
+    when:
+      - osds_dir_stat is defined 
+      - item.stat.exists == false
+
+  - name: install sgdisk(gdisk)
+    package:
+      name: gdisk
+      state: present
+    
+  - name: get osd(s) journal uuid
+    shell: cat "/var/lib/ceph/osd/{{ cluster_name.stdout }}-{{ item.1.osd_id }}/journal_uuid"
+    register: osds_uuid
+    with_subelements:
+      - "{{ dev_ssds }}"
+      - partitions
+
+  - name: make partitions on new ssd
+    shell: >
+      sgdisk --new={{item.item[1].index}}:0:+{{item.item[1].size}} "--change-name={{ item.item[1].index }}:ceph journal" --typecode={{ item.item[1].index }}:{{ journal_typecode }} --partition-guid={{ item.item[1].index }}:{{ item.stdout }} --mbrtogpt -- {{ item.item[0].device_name }}
+    with_items:
+      - "{{ osds_uuid.results }}"
+
+  - name: stop osd(s) service
+    service:
+      name: "ceph-osd@{{ item.item[1].osd_id }}"
+      state: stopped
+      enabled: no
+    with_items:
+      - "{{ osds_uuid.results }}"
+
+  - name: reinitialize osd(s) journal in new ssd
+    shell: >
+       ceph-osd -i {{ item.item[1].osd_id }} --mkjournal --cluster {{ cluster_name.stdout }}
+    with_items:
+      - "{{ osds_uuid.results }}"
+    when: cluster_name is defined
```

Comment:
Hum in some playbook this means including vars, thing that we don't do anymore.
@andrewschoen is it enough to copy the file on the git root dir, run the playbook and assume variable like `{{ cluster }}` will be resolved by `group_vars/all.yml`?
Thanks!